dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIFunctionToolCall.cs

```diff
@@ -84,4 +85,90 @@ public override string ToString()
 
         return sb.ToString();
     }
+
+    /// <summary>
+    /// Tracks tooling updates from streaming responses.
+    /// </summary>
+    /// <param name="update">The tool call update to incorporate.</param>
+    /// <param name="toolCallIdsByIndex">Lazily-initialized dictionary mapping indices to IDs.</param>
+    /// <param name="functionNamesByIndex">Lazily-initialized dictionary mapping indices to names.</param>
+    /// <param name="functionArgmentBuildersByIndex">Lazily-initialized dictionary mapping indices to arguments.</param>
+    internal static void TrackStreamingToolingUpdate(
+        StreamingToolCallUpdate? update,
+        ref Dictionary<int, string>? toolCallIdsByIndex,
+        ref Dictionary<int, string>? functionNamesByIndex,
+        ref Dictionary<int, StringBuilder>? functionArgmentBuildersByIndex)
+    {
+        if (update is null)
+        {
+            // Nothing to track.
+            return;
+        }
+
+        // If we have an ID, ensure the index is being tracked. Even if it's not a function update,
+        // we want to keep track of it so we can send back an error.
+        if (update.Id != null)
+        {
+            (toolCallIdsByIndex ??= new())[update.ToolCallIndex] = update.Id;
+        }
+
+        if (update is StreamingFunctionToolCallUpdate ftc)
+        {
+            // Ensure we're tracking the function's name.
+            if (ftc.Name != null)
+            {
+                (functionNamesByIndex ??= new())[ftc.ToolCallIndex] = ftc.Name;
+            }
+
+            // Ensure we're tracking the function's arguments.
+            if (ftc.ArgumentsUpdate != null)
+            {
+                if (!(functionArgmentBuildersByIndex ??= new()).TryGetValue(ftc.ToolCallIndex, out StringBuilder? argBuilder))
+                {
+                    functionArgmentBuildersByIndex[ftc.ToolCallIndex] = argBuilder = new();
+                }
+
+                argBuilder.Append(ftc.ArgumentsUpdate);
+            }
+        }
+    }
+
+    // TODO: Remove this temporary hack once the Azure SDK has been updated to set `base.Type = "function"` in the ChatCompletionsFunctionToolCall ctor
```

Comment:
@trrwilson, am I correct that this is a bug in Beta 12? If not, how should I be changing this code based on what's exposed in the Azure SDK?